src/apphosting/yaml.ts

```diff
@@ -48,43 +48,26 @@ export class AppHostingYamlConfig {
     return new AppHostingYamlConfig();
   }
 
-  private constructor() {
-    this._environmentVariables = new Map();
-    this._secrets = new Map();
-  }
-
-  get environmentVariables(): EnvironmentVariable[] {
-    return mapToArray(this._environmentVariables);
-  }
-
-  get secrets(): Secret[] {
-    return mapToArray(this._secrets);
-  }
-
-  addEnvironmentVariable(env: EnvironmentVariable) {
-    this._environmentVariables.set(env.variable, env);
-  }
-
-  addSecret(secret: Secret) {
-    this._secrets.set(secret.variable, secret);
-  }
-
-  clearSecrets() {
-    this._secrets.clear();
-  }
-
   /**
    * Merges this AppHostingYamlConfig with another config, the incoming config
    * has precedence if there are any conflicting configurations.
    * */
-  merge(other: AppHostingYamlConfig) {
-    for (const [key, value] of other._environmentVariables) {
-      this._environmentVariables.set(key, value);
+  merge(other: AppHostingYamlConfig, allowSecretsToBecomePlaintext: boolean = true) {
+    if (!allowSecretsToBecomePlaintext) {
+      const wereSecrets = Object.entries(this.env)
+        .filter(([, env]) => env.secret)
+        .map(([key]) => key);
+      if (wereSecrets.some((key) => other.env[key]?.value)) {
+        throw new FirebaseError(
+          `Cannot convert secret to plaintext in ${other.filename ? other.filename : "apphosting yaml"}`,
```

Comment:
```suggestion
          `Cannot convert secret to plaintext in ${other.filename ? other.filename : "apphosting.yaml"}`,
```

Reply:
alternatively, `other.filename ?? "apphosting.yaml"`

Reply:
This shoudl only ever happen in tests where we start with `.empty()` instead of loading from a file. I wanted something different so that we didn't accidentally start printing invalid file names if we didn't actually know what was right.